test/test_fx.py

```diff
@@ -1597,6 +1597,42 @@ def test_no_mutation(self):
         with self.assertRaisesRegex(NotImplementedError, "new_args"):
             x[0] = 4
 
+    def test_error_reporting_after_recompile(self):
+        try:
+            from io import StringIO
+            from unittest.mock import patch
+        except ImportError:
+            self.skipTest("Necessary modules not available")
+
+        class M(torch.nn.Module):
+            def __init__(self):
+                super(M, self).__init__()
+                self.W = torch.nn.Parameter(torch.randn(5))
+
+            def forward(self, x):
+                return torch.dot(self.W, x)
+
+        traced = torch.fx.symbolic_trace(M())
+
+        out = [n for n in list(traced.graph.nodes) if n.op == "output"]
+        relu_out = traced.graph.call_method(method_name='relu', args=(out,))
+        traced.graph.output(relu_out)
+
+        traced.recompile()
+
+        with self.assertRaises(TypeError):
+            traced(5)
+
+        traced(5)
+
+        with patch("sys.stderr", new=StringIO()) as fake_err:
```

Comment:
`TestFX` inherits from `JitTestCase`, which defines a `capture_stdout` context manager: https://github.com/pytorch/pytorch/blob/18a7ec7d7d0a6726e73dc53a6ee1044b8f2513b1/torch/testing/_internal/jit_utils.py#L100

Maybe you can add a similar `capture_stderr` manager on `JitTestCase` and use that here